bazel/private/prebuilt_protoc_extension.bzl

```diff
@@ -0,0 +1,18 @@
+"Module extensions for use under bzlmod"
+
+load("//toolchain:platforms.bzl", "PROTOBUF_PLATFORMS")
+load("//bazel/private:prebuilt_protoc_toolchain.bzl", "prebuilt_protoc_repo")
+
+def create_all_toolchain_repos(name, version):
+    for platform in PROTOBUF_PLATFORMS.keys():
+        prebuilt_protoc_repo(
+            # We must replace hyphen with underscore to workaround rules_python py_proto_library constraint
+            name = ".".join([name, platform.replace("-", "_")]),
+            platform = platform,
+            version = version,
+        )
+
+protoc = module_extension(
```

Comment:
Use skylibs `modules.as_extension` so that this is marked reproducible and you get `bazel mod tidy` fixes for `use_repo` while developing protobuf.